### PR TITLE
BM8 comparison page using Jupyter

### PR DESCRIPTION
**Pull request by wd15**
**Start the switch to using Jupyter notebooks for comparison pages**

Address #1311, #1310, #1298, #1208

- [x] switch to using github actions
- [x] implement BM8a as a notebook
- [x] get a simple nix build working with github actions using cachix
- [x] implement pfhub package
- [x] create a default.nix
- [x] doctests
- [x] linting
- [x] make github actions test package
- [x] test results notebooks

**Remember to tear down the live site when merging or closing this
pull request.**

---
This change is [Reviewable](https://reviewable.io/reviews/usnistgov/pfhub/1314)

### CODE REVIEW

**Comment by pfhub** [comment]
The new [PFHub live website](https://random-cat-1314.surge.sh) is ready for review.

**Comment by pfhub** [comment]
The new [PFHub live website](https://random-cat-1314.surge.sh) is ready for review.

**Comment by pfhub** [comment]
The new [PFHub live website](https://random-cat-1314.surge.sh) is ready for review.

**Comment by pfhub** [comment]
The new [PFHub live website](https://random-cat-1314.surge.sh) is ready for review.

**Comment by pfhub** [comment]
The new [PFHub live website](https://random-cat-1314.surge.sh) is ready for review.

**Comment by pfhub** [comment]
The new [PFHub live website](https://random-cat-1314.surge.sh) is ready for review.

**Comment by pfhub** [comment]
The new [PFHub live website](https://random-cat-1314.surge.sh) is ready for review.

**Comment by pfhub** [comment]
The new [PFHub live website](https://random-cat-1314.surge.sh) is ready for review.

**Comment by pfhub** [comment]
The new [PFHub live website](https://random-cat-1314.surge.sh) is ready for review.

**Comment by pfhub** [comment]
The new [PFHub live website](https://random-cat-1314.surge.sh) is ready for review.

**Comment by pfhub** [comment]
The new [PFHub live website](https://random-cat-1314.surge.sh) is ready for review.

**Comment by pfhub** [comment]
The new [PFHub live website](https://random-cat-1314.surge.sh) is ready for review.

**Comment by pfhub** [comment]
The new [PFHub live website](https://random-cat-1314.surge.sh) is ready for review.

**Comment by pfhub** [comment]
The new [PFHub live website](https://random-cat-1314.surge.sh) is ready for review.

**Comment by pfhub** [comment]
The new [PFHub live website](https://random-cat-1314.surge.sh) is ready for review.

**Comment by pfhub** [comment]
The new [PFHub live website](https://random-cat-1314.surge.sh) is ready for review.

**Comment by pfhub** [comment]
The new [PFHub live website](https://random-cat-1314.surge.sh) is ready for review.

**Comment by pfhub** [comment]
The new [PFHub live website](https://random-cat-1314.surge.sh) is ready for review.

**Comment by pfhub** [comment]
The new [PFHub live website](https://random-cat-1314.surge.sh) is ready for review.

**Comment by pfhub** [comment]
The new [PFHub live website](https://random-cat-1314.surge.sh) is ready for review.

File: _data/pfhub/setup.cfg [hunk 1]
```diff
@@ -0,0 +1,26 @@
+[metadata]
+name = pfhub
+description-file = README.md
+description = Phase Field Community Hub Python Module
+author = Daniel Wheeler
+author_email = daniel.wheeler2@gmail.com
```
Comment (tkphd) [inline _data/pfhub/setup.cfg:6]: Official duty work product, unofficial contact info?

File: _data/pfhub/setup.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
+#!/usr/bin/env python
+
+"""PyMKS - the materials knowledge system in Python
```
Comment (tkphd) [inline _data/pfhub/setup.py:3]: False, this is not PyMKS.

File: results/main.py [hunk 3]
```diff
@@ -0,0 +1,347 @@
+from toolz.curried import filter as filter_
```
Comment (tkphd) [inline results/main.py:1]: Looks like a lot of code duplicated from `pypfhub/main.py`. Can this be consolidated?

**Review by tkphd — CHANGES_REQUESTED** [review]
Minor adjustments. Have not launch NBReview yet.

**Comment by tkphd** [comment]
I don't have access, but in the Repository Settings you should be able to remove the webhook.

**Comment by pfhub** [comment]
The new [PFHub live website](https://random-cat-1314.surge.sh) is ready for review.

**Comment by pfhub** [comment]
The new [PFHub live website](https://random-cat-1314.surge.sh) is ready for review.

**Comment by pfhub** [comment]
The new [PFHub live website](https://random-cat-1314.surge.sh) is ready for review.

**Comment by pfhub** [comment]
The new [PFHub live website](https://random-cat-1314.surge.sh) is ready for review.

File: _data/pfhub/pfhub/__init__.py [hunk 4]
```diff
@@ -0,0 +1,29 @@
+"""PFHub - Phase Field Community Hub
```
[the remark this answers is not in the document]
Comment (wd15) [reply _data/pfhub/pfhub/__init__.py]: Changed to `python-pfhub` for now. Will break out into separate repository eventually.

File: _data/pfhub/setup.cfg (hunk 1 above)
Comment (wd15) [reply _data/pfhub/setup.cfg]: Fixed.

File: _data/pfhub/setup.py (hunk 2 above)
Comment (wd15) [reply _data/pfhub/setup.py]: Fixed.

File: results/main.py (hunk 3 above)
Comment (wd15) [reply results/main.py]: This was legacy. Deleted now.

**Comment by wd15** [comment]
@tkphd: please take another look. Thanks!

**Comment by pfhub** [comment]
The new [PFHub live website](https://random-cat-1314.surge.sh) is ready for review.

File: Makefile [hunk 5]
```diff
@@ -17,6 +17,11 @@ $(HEXBIN_OUT): $(HEXBIN_IN)
 
 %.ipynb.raw.html: %.ipynb
 	jupyter-nbconvert $< --output $(notdir $@) --to html --template basic
+	if [ "results/"  = "$(dir $@)" ]; then \
+	  echo 'got here'; \
```
Comment (tkphd) [inline Makefile:21]: If this is meant to stay, please replace "got here" with a meaningful message, perhaps including the calling process and the filename.

File: _data/python-pfhub/README.md [hunk 6]
```diff
@@ -0,0 +1,22 @@
+# Nix
```
Comment (tkphd) [inline _data/python-pfhub/README.md:1]: Please include a one-line summary of what python-pfhub is meant to accomplish, at least, insert a top-level `# PFHub Python Package`, and increase the existing headings by a level (e.g., `# Nix` -> `## Nix`).

File: _data/python-pfhub/pfhub/__init__.py [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+"""
+
+
+import os
+
+
+def test(*args):
+    r"""Run all the module tests.
+
+    Equivalent to running ``py.test pymks`` in the base of
```
Comment (tkphd) [inline _data/python-pfhub/pfhub/__init__.py:13]: Not PyMKS.

**Review by tkphd — CHANGES_REQUESTED** [review]
Just a few more minor changes.

**Review by tkphd — COMMENTED** [review]
Used ReviewNB to check notebooks -- looks good!

**Comment by tkphd** [comment]
Rendered web page also looks good. Almost there!

File: Makefile (hunk 5 above)
Comment (wd15) [reply Makefile]: Removed it.

File: _data/python-pfhub/README.md (hunk 6 above)
Comment (wd15) [reply _data/python-pfhub/README.md]: Included short description

File: _data/python-pfhub/pfhub/__init__.py (hunk 7 above)
Comment (wd15) [reply _data/python-pfhub/pfhub/__init__.py]: Removed references to PyMKS

**Comment by pfhub** [comment]
The new [PFHub live website](https://random-cat-1314.surge.sh) is ready for review.

**Review by tkphd — APPROVED** [review]
LGTM. Nice work!

**Pull request merged**